### PR TITLE
FIX : Formatting of Approval Letter 

### LINKED ISSUE #9324: The APPROVAL LETTER is not wrapped with the <PrintPreview> component
**Describe the bug**
The APPROVAL LETTER is not wrapped with the `<PrintPreview>` component

**To Reproduce**

1. Go to 'Resource on sidebar'
2. Click on 'any "all details" button'
3. Click on 'APPROVAL LETTER button'

**Expected behavior**
 Wrap the APPROVAL LETTER with the <PrintPreview> component to provide a better preview experience

**Screenshots**
Image

**Desktop (please complete the following information):**

- OS: [windows]
- Browser [chrome]
- Version [131.0.6778.109]

### PR DESCRIPTION
**Pull request by Rajasimhareddybolla**
Fix approval letter formatting to resolve issue #9324

## Proposed Changes

- Fixes #9324 
- Added PrintPreview Component for the Approval Letter
The updated Approval Letter formatting looks like this:
image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a `ResourcePrintPreview` component for displaying a print preview of resource details.
	- Added a new route for printing resource details, enhancing navigation options.
	- Enhanced the `ResourceDetails` component with improved layout and loading state handling.

- **Bug Fixes**
	- Improved rendering logic to ensure proper display of resource details during data fetching.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
print preview already provides the button to print right? why duplicate again?

also let's switch to using `Button` from `ui` directory instead of `ButtonV2`.

Keep the UI identical to how it's present in other places

**Comment by Rajasimhareddybolla** [comment]
Hi @rithviknishad, 

Thanks for your quick feedback!

1. **Duplicate Print Button**: I understand the issue of the duplicate print button due to the existing `PrintPreview` component. To resolve this, I've implemented conditional rendering to hide the redundant print button when the `PrintPreview` component is available.

2. **Button Component**: As requested, I’ve switched to using the primary variant of the `Button` component from the `ui` directory instead of `ButtonV2`.

Here’s the updated outcome:

Updated Approval Letter Screenshot

Please let me know if any further changes are required.

**Review by modamaan — CHANGES_REQUESTED** [review]
Screenshot 2024-12-07 142006
it is a printPreview wrapping issue. Only the `PrintPreview` component's print button is needed. Remove the `ResourceDetails` print approval letter and close button

Reference
https://github.com/ohcnetwork/care_fe/pull/8954

**Review by modamaan — CHANGES_REQUESTED** [review]
We need to solve this without making changes to the PrintPreview reusable component

**Comment by Rajasimhareddybolla** [comment]
Without adding the hideControls prop, the breadcrumb navigation bar appears duplicated. If it's fine, we can remove the 'Resource Details' navigation and use the 'PrintPreview' one instead. Let me know your thoughts!

**Comment by modamaan** [comment]
@Rajasimhareddybolla  yes

**Comment by Rajasimhareddybolla** [comment]
@modamaan I believe we can't remove the duplicate breadcrumb without adding an additional prop because the Page component is creating both the title and breadcrumb. Since the Page component is used to render both 'Resource Details' and 'Approval Letter', we might need to either add a prop to control this behavior or create an entirely new component specifically for PrintPreview to avoid duplication.
```
<Page
  title={"Resource details"}
  crumbsReplacements={{ [props.id]: { name: data.title } }}
  backUrl={"/resource/board"}
>
  {isPrintMode ? (
    <div className="my-4">
      <PrintPreview title="Approval Letter">
        {ApprovalLetter(data)}
      </PrintPreview>
    </div>
  ) : (
    // your other component here
  )}
</Page>
```
Please check this .

**Comment by modamaan** [comment]
Don't want to remove it, try like this. 
Screenshot_20241207_165632_Chrome

**Review by modamaan — APPROVED** [review]
@rithviknishad  LGTM

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Let's keep the print preview in a separate route of it's own. When hitting the back button from the breadcrumbs while in print mode, the we are going back to the resources list page instead of the Resource Details page.

**Comment by Rajasimhareddybolla** [comment]
@rithviknishad, as you mentioned, I have created a new route for the PrintPreview such that:

The back button now navigates to the Resource Details page instead of the Resources List page.
This ensures that the PrintPreview is in a separate route of its own, aligning with the expected behavior.

**Comment by nihal467** [comment]
@rithviknishad can you look into it

File: src/components/Resource/ResourceDetails.tsx [hunk 1]
```diff
@@ -53,321 +51,172 @@ export default function ResourceDetails(props: { id: string }) {
...
+      title={"Resource details"}
+      crumbsReplacements={{ [props.id]: { name: data.title } }}
+      backUrl={"/resource/board"}
+    >
+      <div className="mx-3 mb-10 md:mx-8">
+        <div className="my-4 flex flex-col items-start md:flex-row md:items-center md:justify-between">
+          <ButtonV2 href={`/resource/${props.id}/print`}>
+            <CareIcon icon="l-file-alt" className="mr-2 text-lg" /> Approval
+            Letter
+          </ButtonV2>
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Resource/ResourceDetails.tsx:69]: Switch out the button as coderabbit suggested.

File: src/components/Resource/ResourceDetails.tsx [hunk 2]
```diff
@@ -53,321 +51,172 @@ export default function ResourceDetails(props: { id: string }) {
...
+        </div>
+        {data.assigned_to_object && (
+          <div className="relative rounded-lg bg-primary-200 shadow">
+            <div className="mx-auto max-w-screen-xl p-3 sm:px-6 lg:px-8">
+              <div className="pr-16 sm:px-16 sm:text-center">
+                <p className="font-bold text-primary-800">
+                  <span className="inline">
+                    Assigned to: {formatName(data.assigned_to_object)} -{" "}
```
Comment (Jacobjeevan) [inline src/components/Resource/ResourceDetails.tsx:77]: Let's add translations for all these.

Search for existing key-value pairs in en.json, if not add them, and then use useTranslation hook.

File: src/components/Resource/ResourceDetails.tsx [hunk 3]
```diff
@@ -53,321 +51,172 @@
...
+        <div className="mt-4 h-full rounded-lg border bg-white p-4 text-black shadow">
+          <div className="mb-4 flex flex-col sm:flex-row sm:justify-between">
+            <div className="text-xl font-semibold">{data.title || "--"}</div>
+            <ButtonV2
+              data-testid="update-status"
+              className="mt-4 sm:mt-2"
+              href={`/resource/${data.id}/update`}
+            >
+              Update Status/Details
+            </ButtonV2>
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Resource/ResourceDetails.tsx:94]: Switch out the button here as well.

File: src/components/Resource/ResourcePrintPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+              <div>
+                {data.origin_facility_object?.state_object?.name || "--"}
+              </div>
+            </div>
+            {data.status === "REJECTED" ||
+            data.status === "PENDING" ||
+            data.status === "ON HOLD" ? (
+              <div className="mt-4">
+                <span className="leading-relaxed">
+                  The request for resource (details below) placed by yourself is{" "}
```
Comment (Jacobjeevan) [inline src/components/Resource/ResourcePrintPreview.tsx:70]: Let's get translations added for all these as well.

File: src/components/Resource/ResourcePrintPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+              </span>
+              {data.title || "--"}
+            </div>
+            <div className="mt-1">
+              <span className="font-semibold leading-relaxed">
+                Description of Request:{" "}
+              </span>
+              {data.reason || "--"}
+            </div>
+            <div className="mt-4">
```
Comment (Jacobjeevan) [inline src/components/Resource/ResourcePrintPreview.tsx:105]: Since these styles are reoccurring, let's create a common fn that renders them.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@Rajasimhareddybolla what's the status of the PR

**Pull request closed without merging**